lib/ts/controllers/s-navigation-tablist.ts

```diff
@@ -0,0 +1,121 @@
+namespace Stacks {
+    export class TabListController extends Stacks.StacksController {
+
+        private boundSelectTab: any; // (event: MouseEvent) => void;
+        private boundHandleKeydown: any // (event: KeyboardEvent) => void;
+
+        connect() {
+            super.connect();
+
+            this.boundSelectTab = this.selectTab.bind(this);
+            this.boundHandleKeydown = this.handleKeydown.bind(this);
+
+            for (let tab of this.tabTargets) {
+                tab.addEventListener("click", this.boundSelectTab);
+                tab.addEventListener("keydown", this.boundHandleKeydown);
+            }
+        }
+
+        disconnect() {
+            super.disconnect();
+
+            for (let tab of this.tabTargets) {
+                tab.removeEventListener("click", this.boundSelectTab);
+                tab.removeEventListener("keydown", this.boundHandleKeydown);
+            }
+        }
+
+        /**
+         * Gets all tabs within the controller.
+         */
+        get tabTargets() {
+            return <HTMLElement[]>Array.from(this.element.querySelectorAll("[role=tab]"));
+        }
+
+        /**
+         * Handles click events on individual tabs, causing them to be selected.
+         */
+        selectTab(event: MouseEvent) {
+            this.switchToTab(<HTMLElement>event.currentTarget);
+
+            // We need to be very aggressive about selecting the current tab on click.
+            // Mouse click will not always achieve this and failing to do so will prevent
+            // keyboard navigation from working.
+            this.selectedTab?.focus();
+        }
+
+        /**
+         * Handles left and right arrow keydown events on individual tabs,
+         * selecting the adjacent tab corresponding to the event.
+         */
+        handleKeydown(event: KeyboardEvent) {
+            let tabElement = <HTMLElement>event.currentTarget;
+
+            var tabs = this.tabTargets;
+            var tabIndex = tabs.indexOf(tabElement);
+    
+            if (event.key === "ArrowRight") {
+                tabIndex++;
+            } else if (event.key === "ArrowLeft") {
+                tabIndex--;
+            } else {
+                return;
+            }
+    
+            if (tabIndex < 0) { tabIndex = tabs.length - 1; }
+            if (tabIndex >= tabs.length) { tabIndex = 0; }
```

Comment:
I love that you thought through and did this. The circular selection feels really good in the implementation. I'd recommend adding a very succinct comment explaining:

```suggestion
            // if the user selection goes off either end of the array, loop back around to the other side
            if (tabIndex < 0) { tabIndex = tabs.length - 1; }
            if (tabIndex >= tabs.length) { tabIndex = 0; }
```